sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/SocketClient.java

```diff
@@ -0,0 +1,182 @@
+package com.azure.core.http.httpurlconnection;
+
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.httpurlconnection.implementation.HttpUrlConnectionResponse;
+import reactor.core.publisher.Flux;
+
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+import java.io.*;
+import java.net.ProtocolException;
+import java.net.Socket;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.util.*;
+
+/**
+ * A socket client used for making PATCH requests
+ */
+
+class SocketClient {
+
+    private final String host;
+    private final int port;
+    private final String username;
+    private final String password;
```

Comment:
If SocketClient was stateless (i.e. it did not have these fields) you could simplify the code in the client - you would no longer need to have a map of SocketClient instances for different host/port combinations.